M2/Macaulay2/packages/HomologicalShiftIdeals.m2

```diff
@@ -0,0 +1,949 @@
+    Example
+      S = QQ[x_1..x_4];
+      I = ideal(x_1^2,x_2,x_3,x_1^2*x_4);
+      supportIdeal(I)
+    Example
+      S = QQ[x_1..x_4];
+      I = ideal(x_1^2,x_2^5,x_3,x_1*x_4);
+      supportIdeal(I)
+///);
+
+supportIdeal = method(TypicalValue=>List)
+supportIdeal (Ideal) := I -> (
+        if not isMonomialIdeal(I) then error "expected a monomial ideal";
+        S:=ring I;
+        GI:=flatten entries mingens I;
+        suppI:=set{};
+        sum(GI, w -> set support w)
+    );
+
+-------------------------------------------------------------------------------------------
+-- Whether an ideal is fully supported
+-------------------------------------------------------------------------------------------
+mydoc = concatenate(mydoc,///
+Node
+  Key
+    isFullySupported
+    (isFullySupported,Ideal)
+  Headline
+    whether a monomial ideal is fully supported
+  Usage
+    isFullySupported(I)
+  Inputs
+    I:Ideal
+      a monomial ideal $I$ of a polynomial ring $S = K[x_1,\ldots,x_n]$, $K$ a field
+  Outputs
+    b:Boolean
+      whether $I$ is fully supported
+  Description
+    Text
+      A monomial ideal $I$ of a polynomial ring $S=K[x_1,\ldots,x_n]$ is fully supported if $\text{supp}(I)=\{x_1,\dots,x_n\}$.
+    Example
+      S = QQ[x_1..x_4];
+      I = ideal(x_1^2,x_2,x_3,x_1^2*x_4);
+      isFullySupported(I)
+    Example
+      S = QQ[x_1..x_4];
+      I = ideal(x_1^2,x_3,x_2*x_4);
+      isFullySupported(I)
+///);
+
+isFullySupported = method(TypicalValue=>Boolean)
+isFullySupported (Ideal) := I -> (
+        n:=numgens ring I;
+        suppI:=supportIdeal I;
+        #suppI==n
+    );
+
+-------------------------------------------------------------------------------------------
+--  Compute the monomial with a given multidegree
+-------------------------------------------------------------------------------------------
+mydoc = concatenate(mydoc,///
+Node
+  Key
+    toMonomial
+    (toMonomial,Ring,List)
+  Headline
+    compute the monomial with a given multidegree
+  Usage
+    toMonomial(S,l)
+  Inputs
+    S:Ring
+      a polynomial ring $S = K[x_1,\ldots,x_n]$, $K$ a field
+    l:List
+      the multidegree $(a_1,\dots,a_n)$
+  Outputs
+    u:RingElement
+      the monomial $x_1^{a_1}\cdots x_n^{a_n}\in S$
+  Description
+    Text
+      The multidegree of a monomial $u\in S=K[x_1,\ldots,x_n]$ is the vector $(a_1,a_2,\dots,a_n)$ where $a_i=\max\{j:x_i^j\ \text{divides}\ u\}$.
+    Example
+      S = QQ[x_1..x_4];
+      l = {0,2,1,2};
+      toMonomial(S,l)
+///);
+
+toMonomial = method(TypicalValue=>RingElement)
+toMonomial (Ring,List) := (S,l) -> (
+	n:=numgens S-1;
+	u:=1;
+        j:=0;
+        if #l != n+1 then error "the list does not have the right lenght";
```

Comment:
This line has a typo ("lenght" -> "length")

Reply:
Shall I fix and review the change?

Reply:
You pushed a commit fixing the typo, and then pushed another commit changing it back to the misspelling!  :)

Reply:
Oh, I am sorry! So I wait for your instructions before making more mess.

Reply:
No worries!  You can just push another commit changing it back to "length"